tools/ci_build/exclude_unused_ops_and_types.py

```diff
@@ -41,22 +52,35 @@ def _should_exclude_op(self, domain, operator, start_version, end_version):
         return True
 
     def process_registration(self, lines: typing.List[str], constant_for_domain: str, operator: str,
-                             start_version: int, end_version: int = None, type: str = None):
+                             start_version: int, end_version: typing.Optional[int] = None,
+                             type: typing.Optional[str] = None):
         # convert from the ORT constant name to the domain string used in the config
         domain = op_registration_utils.map_ort_constant_to_domain(constant_for_domain)
         exclude = False
+        exclude_reason = ""
 
         if domain:
             # see if entire op is excluded
-            exclude = self._should_exclude_op(domain, operator, start_version, end_version)
+            if self._should_exclude_op(domain, operator, start_version, end_version):
+                exclude_reason = "Entire op is excluded."
+                exclude = True
 
             # see if a specific typed registration can be excluded
-            if not exclude and type and self._op_types_usage_manager:
-                exclude = not self._op_types_usage_manager.is_typed_registration_needed(domain, operator, type)
+            if not exclude and type and self._op_type_usage_manager \
+                    and not self._op_type_usage_manager.is_typed_registration_needed(domain, operator, type):
+                exclude_reason = "Specific typed registration is excluded by OperatorTypeUsageManager."
+                exclude = True
+
+        # see if a valid type can be excluded because it is not one of the globally allowed types
+        if not exclude and type and self._globally_allowed_types and type in _valid_allowed_types \
+                and type not in self._globally_allowed_types:
```

Comment:
In order to handle the OneHot and Dropout registrations do we need to pass through the globally allowed types to the TypeUsageProcessor so we can plug in op specific logic there? Or should we just special case them here with a function that checks for X matches of allowed types in the registration string (3 matches for OneHot and 2 for Dropout)?

Reply:
Re-did this a bit. Not using the TypeUsageProcessor anymore, but rather a separate RegistrationProcessor. It should handle OneHot now.